pkg/kubectl/custom_column_printer.go

```diff
@@ -35,6 +37,95 @@ const (
 	flags             = 0
 )
 
+// MassageJSONPath attempts to be flexible with JSONPath expressions, it accepts:
+//   * metadata.name (no leading '.' or curly brances '{...}'
+//   * {metadata.name} (no leading '.')
+//   * .metadata.name (no curly braces '{...}')
+//   * {.metadata.name} (complete expression)
+// And transforms them all into a valid jsonpat expression:
+//   {.metadata.name}
+func massageJSONPath(pathExpression string) string {
```

Comment:
This seems like it would be better implemented with a regex, so you can do stricter checking and return an error if it doesn't match the set of accepted expressions. I think current implementation will barf somewhere deep in CustomColumnsPrinter if you pass it expressions like `.metadata.name}`, `{.metadata.name`